tests/testthat/test-independent-testing-prepare_ae_summary.R

```diff
@@ -260,4 +263,777 @@ test_that("with one or more adverse event count", {
 
   # name matches
   expect_equal(x$name[2], "with one or more adverse events")
-})
\ No newline at end of file
+})
+
+
+test_that("with one or more adverse event count", {
+
+  adae_itt <- full_join(adsl %>% select(USUBJID, TRT01AN, ITTFL), # merge with adsl to get percentage
+                        adae,
+                        by = "USUBJID"
+  ) %>% subset(ITTFL == "Y" & TRTEMFL == "Y")
+
+  adae_tot <- adae_itt %>% mutate(TRT01AN = 99) # add total rows into calculation
+  adae_tot <- rbind(adae_itt, adae_tot)
+
+  res_ae <- adae_tot %>%
+    group_by(TRT01AN, ITTFL) %>%
+    summarise(n = n_distinct(USUBJID), .groups = "drop")
+
+  res_no <- full_join(res_tot1, # merge with adsl to get percentage
+                        res_ae,
+                        by = c("TRT01AN","ITTFL")) %>% mutate(p=z-n)
+
+  res_no <- res_no %>%
+    mutate(pct = formatC(100 * p / z, digits = 13, format = "f", flag = "0"))
+
+  res_no$pct <- as.numeric(res_no$pct)
+
+  res_no <- res_no %>%
+    mutate(x = paste0("n_", row_number()),
+           prop = paste0("prop_", row_number()))
+
+  res_ae3 <- res_no %>%
+            pivot_wider(
+           id_cols = "ITTFL",
+           names_from = "x",
+           values_from = p,
+           values_fill = list(p = 0)
+         )
+
+  res_prop3 <- res_no %>%
+    pivot_wider(
+      id_cols = "ITTFL",
+      names_from = "prop",
+      values_from = pct,
+      values_fill = list(pct = 0)
+    )
+
+  res_diff3 <- res_prop3 %>% mutate(
+    diff_2 = as.numeric(prop_2) - as.numeric(prop_1),
+    diff_3 = as.numeric(prop_3) - as.numeric(prop_1)
+  )
+  # n count
+  expect_equal(xx %>% filter(x == 3) %>% select(-c("x$name", x)), res_ae3 %>% select(-c(ITTFL)))
```

Comment:
Hi @ginnaram13 , could you please confirm if the derivation of `res_ae3` follows the same logic as the SAS macro?

Reply:
Yes, after confirming with Madhu.